Library/Homebrew/cmd/update-report.rb

```diff
@@ -101,6 +104,32 @@ def install_core_tap_if_necessary
     ENV["HOMEBREW_UPDATE_BEFORE_HOMEBREW_HOMEBREW_CORE"] = revision
     ENV["HOMEBREW_UPDATE_AFTER_HOMEBREW_HOMEBREW_CORE"] = revision
   end
+
+  def migrate_legacy_cache_if_necessary
```

Comment:
For the same volume check: We also have [`Homebrew::Diagnostic::Volumes`](https://github.com/Homebrew/brew/blob/adc293637f58993bb52792d664cc38bf865eeab4/Library/Homebrew/diagnostic.rb#L31-L70) that is used by a few doctor checks.